tests/by-util/test_cp.rs

```diff
@@ -1536,6 +1538,37 @@ fn test_copy_through_dangling_symlink_no_dereference() {
         .no_stdout();
 }
 
+/// Test for copying a dangling symbolic link and its permissions.
+#[test]
+fn test_copy_through_dangling_symlink_no_dereference_permissions() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    //               target name    link name
+    at.symlink_file("no-such-file", "dangle");
+    //          don't dereference the link
+    //           |    copy permissions, too
+    //           |      |    from the link
+    //           |      |      |     to new file d2
+    //           |      |      |        |
+    //           V      V      V        V
+    ucmd.args(&["-P", "-p", "dangle", "d2"])
```

Comment:
excellent doc, bravo

Reply:
I was having a hard time remembering the meaning of each argument, so I needed to write it down :)